src/App.tsx

```diff
@@ -1,223 +1,224 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { Todo } from './types/Todo';
+import { FilterValues } from './types/FilterValues';
+import {
+  deleteTodo,
+  getTodos,
+  patchTodo,
+  postTodo,
+} from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterValue, setFilterValue] = useState(FilterValues.ALL);
+  const [title, setTitle] = useState('');
+  const [isTodoAdded, setIsTodoAdded] = useState(false);
+  const [selectedTodosIds, setSelectedTodosIds] = useState<number[]>([]);
+
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+  }, [isTodoAdded]);
+
+  useEffect(() => {
+    const getTodosAsync = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch {
+        setErrorMessage('Unable to load todos');
+      }
+    };
+
+    if (user) {
+      getTodosAsync(user.id);
+    }
+  }, []);
+
+  const hundleAddTodo = useCallback(async (inputTitle: string) => {
```

Comment:
can you move it to Header? App will be easier )

Reply:
Yes, I understand it) But I would like to have all three methods(add, delete, update) in one file. Besides it will be necessary to add many more props to the Header. But if you insist, I will do it)))

Reply:
I moved `UseEffect` with `newTodoField` to Header))